service/src/main/java/org/mskcc/cmo/metadb/service/impl/RequestReplyHandlingServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.messaging.MessageConsumer;
+import org.mskcc.cmo.metadb.model.SampleMetadata;
+import org.mskcc.cmo.metadb.service.RequestReplyHandlingService;
+import org.mskcc.cmo.metadb.service.SampleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.nats.client.Message;
+
+@Component
+public class RequestReplyHandlingServiceImpl implements RequestReplyHandlingService {
+    
+    @Value("${request_reply.patient_samples_topic}")
+    private String REQUEST_REPLY_TOPIC;
+
+    @Value("${num.new_request_handler_threads}")
+    private int NUM_NEW_REQUEST_HANDLERS;
+
+    @Autowired
+    private SampleService sampleService;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private static boolean initialized = false;
+    private static volatile boolean shutdownInitiated;
+    private static final ExecutorService exec = Executors.newCachedThreadPool();
+    private static final BlockingQueue<ReplyInfo> newRequestReplyQueue =
+        new LinkedBlockingQueue<ReplyInfo>();
+    private static CountDownLatch newRequestHandlerShutdownLatch;
+    private static Gateway messagingGateway;
+    
+    private static final Log LOG = LogFactory.getLog(RequestReplyHandlingServiceImpl.class);
+
+    
+    private class ReplyInfo {
+        String replyMessage;
+        String replyTo;
+        
+        ReplyInfo(String replyMessage, String replyTo) {
+            this.replyMessage = replyMessage;
+            this.replyTo = replyTo;
+        }
+        
+        String getReplyMessage() {
+            return replyMessage;
+        }
+        
+        String getReplyTo() {
+            return replyTo;
+        }
+    }
+    
+    private class NewRequestReplyHandler implements Runnable {
+
+        final Phaser phaser;
+        boolean interrupted = false;
+
+        NewRequestReplyHandler(Phaser phaser) {
+            this.phaser = phaser;
+        }
+
+        @Override
+        public void run() {
+            phaser.arrive();
+            while (true) {
+                try {
+                    ReplyInfo replyInfo = newRequestReplyQueue.poll(100, TimeUnit.MILLISECONDS);
+                    if ((replyInfo != null)) {
+                        System.out.println("Publishing to: " + replyInfo.getReplyTo() + "\n message: "+ replyInfo.getReplyMessage());
```

Comment:
Lets get rid of the output to stdout. If we want it, we should put into a log file.